src/Features/Core/Portable/Snippets/ConvertToLSPSnippetService.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Snippets
+{
+    internal class ConvertToLSPSnippetService : ILanguageService
+    {
+        public string? GenerateLSPSnippet(TextChange textChange, List<(string, List<TextSpan>)> placeholders)
+        {
+            var textChangeText = textChange.NewText!;
+
+            for (var i = 0; i < placeholders.Count; i++)
+            {
+                var (identifier, placeholderList) = placeholders[i];
+                if (identifier.Length != 0)
+                {
+                    var newStr = $"${{{i}:{identifier}}}";
+                    textChangeText = textChangeText.Replace(identifier, newStr);
```

Comment:
Is it okay to just assume a Replace will work? If not, is there a better way to modify the string in place?

Reply:
what semantics do you want here?  can you document (on the method) what hte input data is, and what you're transforming it to?

in general .Replace is not going to be safe here as there may be N copies of htat identifier in the change.